com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage

_master_ (commit 55958e30a94d419f740be754876f94fc8dc58f1f)

Failed on IBM JDK8: http://jenkins.hazelcast.com/job/jet-oss-master-ibm-jdk8/248/testReport/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_nodeDown_then_jobRestartsFromSnapshot_twoStage/

Stacktrace:
```
java.lang.AssertionError: No snapshot produced
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.core.JetTestSupport.lambda$waitForFirstSnapshot$3(JetTestSupport.java:253)
	at com.hazelcast.jet.core.JetTestSupport$$Lambda$1172.00000000AC3466B0.run(Unknown Source)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.JetTestSupport.waitForFirstSnapshot(JetTestSupport.java:250)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot(JobRestartWithSnapshotTest.java:204)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage(JobRestartWithSnapshotTest.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:818)
```

Standard output:
```
Hiccups measured while running test 'when_nodeDown_then_jobRestartsFromSnapshot_twoStage(com.hazelcast.jet.core.JobRestartWithSnapshotTest):'
01:56:55, accumulated pauses: 22 ms, max pause: 6 ms, pauses over 1000 ms: 0
01:57:00, accumulated pauses: 497 ms, max pause: 94 ms, pauses over 1000 ms: 0
01:57:05, accumulated pauses: 638 ms, max pause: 50 ms, pauses over 1000 ms: 0
01:57:10, accumulated pauses: 647 ms, max pause: 254 ms, pauses over 1000 ms: 0
01:57:15, accumulated pauses: 208 ms, max pause: 172 ms, pauses over 1000 ms: 0
```

Test `com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_singleStage` from the same test case failed in another build. It seems to be the same issue: http://jenkins.hazelcast.com/job/jet-oss-master-ibm-jdk8/249/testReport/junit/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_nodeDown_then_jobRestartsFromSnapshot_singleStage/

Stacktrace:
```
java.lang.AssertionError: No snapshot produced
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.core.JetTestSupport.lambda$waitForFirstSnapshot$3(JetTestSupport.java:253)
	at com.hazelcast.jet.core.JetTestSupport$$Lambda$1156.0000000068316890.run(Unknown Source)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.JetTestSupport.waitForFirstSnapshot(JetTestSupport.java:250)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot(JobRestartWithSnapshotTest.java:204)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_singleStage(JobRestartWithSnapshotTest.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:818)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetTestSupport.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetTestSupport.java
@@ -267,7 +267,7 @@ public abstract class JetTestSupport extends HazelcastTestSupport {
             JobExecutionRecord record = jr.getJobExecutionRecord(jobId);
             assertNotNull("jobExecutionRecord is null", record);
             snapshotId[0] = record.snapshotId();
-            assertTrue("No more snapshots produced after restart in " + timeoutSeconds + " seconds",
+            assertTrue("No more snapshots produced in " + timeoutSeconds + " seconds",
                     snapshotId[0] > originalSnapshotId);
             assertTrue("stats are 0", allowEmptySnapshot || record.snapshotStats().numBytes() > 0);
         }, timeoutSeconds);

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java
@@ -199,7 +199,7 @@ public class JobRestartWithSnapshotTest extends JetTestSupport {
         Job job = instance1.newJob(dag, config);
 
         JobRepository jobRepository = new JobRepository(instance1);
-        int timeout = (int) (MILLISECONDS.toSeconds(config.getSnapshotIntervalMillis() * 3) + 2);
+        int timeout = (int) (MILLISECONDS.toSeconds(config.getSnapshotIntervalMillis() * 7) + 2);
 
         waitForFirstSnapshot(jobRepository, job.getId(), timeout, false);
         waitForNextSnapshot(jobRepository, job.getId(), timeout, false);

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobRestartWithSnapshotTest.java
@@ -199,7 +199,7 @@ public class JobRestartWithSnapshotTest extends JetTestSupport {
         Job job = instance1.newJob(dag, config);
 
         JobRepository jobRepository = new JobRepository(instance1);
-        int timeout = (int) (MILLISECONDS.toSeconds(config.getSnapshotIntervalMillis() * 7) + 2);
+        int timeout = (int) (MILLISECONDS.toSeconds(config.getSnapshotIntervalMillis() * 3) + 8);
 
         waitForFirstSnapshot(jobRepository, job.getId(), timeout, false);
         waitForNextSnapshot(jobRepository, job.getId(), timeout, false);

